.argo-ci/ci.yaml

```diff
@@ -0,0 +1,137 @@
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "make controller-image"
+
+      - name: executor-image
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "make executor-image"
+
+      - name: cli-linux
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "make cli-linux"
+
+      - name: cli-darwin
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "make cli-darwin"
+
+      - name: ui-image
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "make ui-image"
+          - name: dind-mem
+            value: "4096"
+
+      - name: lint
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "dep ensure && make lint"
+
+      - name: test
+        template: ci-builder
+        arguments:
+          parameters:
+          - name: revision
+            value: "{{inputs.parameters.revision}}"
+          - name: repo
+            value: "{{inputs.parameters.repo}}"
+          - name: cmd
+            value: "dep ensure && make test"
+          - name: mem
+            value: "1024"
+
+  - name: ci-builder
+    inputs:
+      parameters:
+      - name: revision
+      - name: repo
+      - name: cmd
+      - name: mem
+        default: "512"
+      - name: dind-mem
+        default: "600"
+      artifacts:
+      - name: code
+        path: /go/src/github.com/argoproj/argo
+        git:
+          repo: "{{inputs.parameters.repo}}"
+          revision: "{{inputs.parameters.revision}}"
+    container:
+      image: argoproj/argo-ci-builder:1.0
+      command: [sh, -c]
+      args: ["until docker ps; do sleep 3; done; cd /go/src/github.com/argoproj/argo && {{inputs.parameters.cmd}}"]
+      resources:
+        mem_mib: "{{inputs.parameters.mem}}"
```

Comment:
In k8s the resources are just `memory` and `cpu`

Reply:
Maybe we should just omit these values since they are hard to guess, and we mostly don't care about limiting it. It will make the yaml simpler

Reply:
Agree. Removed resources altogether. https://github.com/argoproj/argo/pull/619/commits/1d4f14dad682966ebc08cdcb8837913aeb556a24